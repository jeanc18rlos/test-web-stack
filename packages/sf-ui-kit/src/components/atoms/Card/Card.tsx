import { HTMLAttributes, memo } from 'react';
import StyledCard from './styled/StyledCard';
type CardProps = { children: React.ReactNode };

const Card = ({ children, ...rest }: CardProps & HTMLAttributes<HTMLElement>) => (
  <StyledCard {...rest}>{children}</StyledCard>
);

export default memo(Card);
