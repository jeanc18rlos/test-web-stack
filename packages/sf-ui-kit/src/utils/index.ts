export { sizes, Typography } from './typography';
