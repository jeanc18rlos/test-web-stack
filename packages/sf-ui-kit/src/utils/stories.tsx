import styled from 'styled-components';
import { rem } from 'polished';

// Images
import logo from '../assets/superformula.png';
import { FC, HTMLProps, ReactNode } from 'react';

const StyledStoryHeading = styled.div`
  width: 100%;
  max-width: 85vw;
  margin: 2.5rem auto;
  text-transform: capitalize;
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${({ theme }) => theme.fonts.primaryFont};
    color: ${({ theme }) => theme.colors.black};
    @media screen and (max-width: ${rem('768px')}) {
      font-size: ${rem('20px')};
    }
  }
  hr {
    border-color: ${({ theme }) => theme.colors.black};
  }
`;

export const StoryContainer = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  width: 100%;
  > * {
    margin: 1rem;
  }
`;
export const StoryWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  align-items: center;

  th,
  td {
    font-weight: 400;
    font-family: ${({ theme }) => theme.fonts.primaryFont};
    padding: 0.5em 1em;
  }
`;

export const ComponentWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: 'row';
`;

const StyledStoryHeader = styled.div`
  margin: ${rem('8px')};
  width: calc(100% - ${rem('16px')});
  background: transparent;
  display: flex;
  justify-content: space-between;
  padding: 0 ${rem('50px')};
  border-radius: ${rem('8px')} ${rem('8px')} 0 0;
  position: relative;
  &::before {
    content: '';
    z-index: -1;
    border-top: 88px solid #18181a;
    border-left: 20px solid transparent;
    border-right: 20px solid transparent;
    height: 0;
    background: transparent;
    position: absolute;
    width: 100%;
    left: 0;
    top: 0;
    max-width: 100%;
    box-sizing: border-box;
    filter: drop-shadow(2px 2px 2px black);
  }
  img {
    position: absolute;
    right: 0;
    width: 125px;
    filter: drop-shadow(2px 4px 6px black);
  }
  box-sizing: border-box;
  h1 {
    display: flex;
    justify-content: center;
    align-items: center;
    color: white;
    font-family: ${({ theme }) => theme.fonts.primaryFont};
  }
  @media screen and (max-width: ${rem('768px')}) {
    &::before {
    content: '';
    z-index: -1;
    border-top: 50px solid #18181a;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;

  }
    h1 {
      font-size: ${rem('20px')};
    }
    img {
      width: ${rem('70px')};
    }
  }
`;

export const StoryHeading: FC<HTMLProps<ReactNode>> = ({ children }) => (
  <StyledStoryHeading>
    {children}
    <hr />
  </StyledStoryHeading>
);

export const StoryHeader: FC<HTMLProps<ReactNode> & { name: string }> = ({ name }) => (
  <StyledStoryHeader>
    <h1>{name}</h1>
    <img src={logo} />
  </StyledStoryHeader>
);
