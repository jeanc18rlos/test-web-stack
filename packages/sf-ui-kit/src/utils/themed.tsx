import { ReactNode } from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from '../theme';

const themed = (component: ReactNode): renderer.ReactTestRenderer =>
  renderer.create(<ThemeProvider>{component}</ThemeProvider>);

export { themed };
