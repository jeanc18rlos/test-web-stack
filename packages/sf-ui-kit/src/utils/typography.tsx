import { css } from 'styled-components';
import { rem } from 'polished';

export const sizes = {
  h1: {
    size: `${rem('48px')}`,
    height: `${rem('60px')}`,
    weight: 300,
    style: 'normal'
  },
  h2: {
    size: `${rem('21px')}`,
    height: `${rem('26px')}`,
    weight: 600,
    style: 'normal'
  },
  body: {
    size: `${rem('16px')}`,
    height: `${rem('20px')}`,
    weight: 300,
    style: 'normal'
  }
};

enum Base {
  'h1',
  'h2',
  'body'
}

interface Options {
  size?: string;
  height?: number;
  weight?: number | string;
  style?: string;
  spacing?: string;
  color?: string;
  family?: string;
}

export const Typography = ({ base, options }: { base: keyof typeof Base; options?: Options }): any => css`
  font-feature-settings: 'cpsp' on;
  font-size: ${options && options.size ? options.size : sizes[base].size};
  line-height: ${options && options.height ? options.height : sizes[base].height};
  font-weight: ${options && options.weight ? options.weight : sizes[base].weight};
  font-style: ${options && options.style ? options.style : sizes[base].style};
  ${options && options.spacing && `letter-spacing: ${options && options.spacing};`}
  color: ${({ theme }) => (options && options.color ? options.color : theme.colors.black)};
  font-family: ${({ theme }) => (options && options.family ? options.family : theme.fonts.primaryFont)};
`;
