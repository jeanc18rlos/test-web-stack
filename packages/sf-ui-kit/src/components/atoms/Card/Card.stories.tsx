import { Meta } from '@storybook/react';
import Card from './Card';
import Avatar from '../Avatar';
import { StoryHeader, StoryHeading, StoryWrapper, StoryContainer } from '../../../utils/stories';
import oval from '../../../assets/oval.png';

export default {
  title: 'Components/Atoms/Card',
  component: Card
} as Meta;

const UserCard = () => (
  <Card>
    <Avatar image={oval} alt="Superformula Member" />
    <h2>Jessica may</h2>
    <p>lorem ipsum dolor sit amet</p>
  </Card>
);

export const base = () => (
  <StoryWrapper>
    <StoryHeader name={`Card`} />
    <StoryHeading>
      <h2>Default</h2>
    </StoryHeading>
    <StoryContainer>
      <UserCard />
    </StoryContainer>
  </StoryWrapper>
);

export const grid = () => (
  <StoryWrapper>
    <StoryHeader name={`Card`} />
    <StoryHeading>
      <h2>Default</h2>
    </StoryHeading>
    <StoryContainer>
      <UserCard />
      <UserCard />
      <UserCard />
      <UserCard />
      <UserCard />
      <UserCard />
      <UserCard />
    </StoryContainer>
  </StoryWrapper>
);
