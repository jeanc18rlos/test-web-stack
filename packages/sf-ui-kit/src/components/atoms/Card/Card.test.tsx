import {themed} from "../../../utils/themed";
import Card from ".";

describe("Card", () => {
  it("should render correctly", () => {
    const component = themed(
      <Card>
        Test
      </Card>,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
