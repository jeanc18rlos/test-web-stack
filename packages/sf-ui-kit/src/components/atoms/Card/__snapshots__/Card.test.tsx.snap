// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 25rem;
  height: 21rem;
  padding: 1rem;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  min-height: -webkit-fit-content;
  min-height: -moz-fit-content;
  min-height: fit-content;
  background: #ffffff;
  border-radius: 8px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-filter: drop-shadow(0px 0px 0px rgba(0,0,0,0));
  filter: drop-shadow(0px 0px 0px rgba(0,0,0,0));
  -webkit-transition: filter 0.25s ease-in-out;
  transition: filter 0.25s ease-in-out;
}

.c0:hover,
.c0:focus,
.c0.focus,
.c0.hover {
  -webkit-filter: drop-shadow(0px 4px 10px rgba(0,0,0,0.1));
  filter: drop-shadow(0px 4px 10px rgba(0,0,0,0.1));
}

@media (max-width:75em) {
  .c0 {
    width: 20rem;
    height: 16.8125rem;
  }
}

@media (max-width:64em) {
  .c0 {
    width: 17.5rem;
    height: 14.75rem;
  }
}

@media (max-width:48em) {
  .c0 {
    width: 16rem;
    height: 13.5rem;
  }
}

<section
  className="c0"
>
  Test
</section>
`;
